src/ray/rpc/grpc_server.h

```diff
@@ -77,11 +78,26 @@ class GrpcServer {
   /// \param[in] name Name of this server, used for logging and debugging purpose.
   /// \param[in] port The port to bind this server to. If it's 0, a random available port
   ///  will be chosen.
+  ///
+  /// Note: Ideally with C++20 we could use constraints here, or auto&& in the contructor.
+  template <typename T = std::shared_future<ClusterID>,
+            typename = typename std::enable_if_t<
+                std::is_convertible<T, std::shared_future<ClusterID>>::value>>
   GrpcServer(std::string name,
              const uint32_t port,
              bool listen_to_localhost_only,
+             T &&cluster_token_future = T(),
```

Comment:
I feel this is a little bit confuse. Let's just add another constructor for this.

Reply:
Because of the perfect forwarding?

Reply:
I see. I think it's OK, it's just a ClusterID (28bytes) and only one time thing :)